sdks/python/apache_beam/runners/common.py

```diff
@@ -197,7 +197,11 @@ def receive(self, windowed_value):
   def _dofn_simple_invoker(self, element):
     self._process_outputs(element, self.dofn_process(element.value))
 
-  def _dofn_window_invoker(self, element, args, kwargs, window):
+  def _dofn_per_window_invoker(self, element, args, kwargs):
+    if self.has_windowed_inputs:
```

Comment:
this computation is done twice in one of the code path when there are multiple windows

Suggestion:
 - Remove the if condition here
 - Remove the length check as well
 - Revert the function signature to the old one

That I think would work as well, thoughts ?